### ISSUE TITLE
Target and source image are misnomers

### ISSUE THREAD

**Issue by tdewolff** [BioMedIA/IRTK#12]
Is IRTK still being developed on, or has all effort been shifted to MIRTK?

I wanted to state here (perhaps also useful for other users) the fact that the names **target** and **source** for the input images for registration are very confusing. In our research team this has cost us considerable time. I guess we should have better read the (sparse) documentation, as it clearly states "The transformation that is estimated maps locations in the target to locations in the source".

In other words, the transformation data exists in the area of the target image and maps BACK to the source image. So if you have a reference image and a deformed image, and you want to register the transformation from reference to deformed (ie. all transformation data exists in the area of the reference image and maps to the deformed image), then **source = deformed** and **target = reference**. And not the other way around, as one might expect by the naming!

**Comment by schuhschuh** [comment]
Hi, thanks for the feedback! Development of IRTK has seized. I recommend trying out MIRTK.

Regarding target vs source, no matter how you name these,  users will get confused due to the nature of how images are deformed. The current naming is common and makes a lot of sense when you understand the underlying equations and implementation of image deformations. The source image is the source of the intensity values that are being resampled given the mapping from target space to source domain. Images are being "pulled back" when deformed. It gets more confusing when you consider point clouds such as landmarks. In this case you need the inverse mapping that maps source landmarks to target locations. But that direction of the mapping, you don't use for "deforming" the source image.

Think of target as the grid/ regular sampling of your deformed image space which is the target for resampling the source image onto. To give another explanation, if you would change the order, you would end up mapping source image voxels to a irregular grid in the target space and then have a difficult time to intepolate this irregular data in order to sample it regularly again and thus be able to store as common image.

Hope these explanations help.
